src/services/services.ts

```diff
@@ -992,9 +994,13 @@ namespace ts {
             return !isString(entry) ? entry : undefined;
         }
 
-        public getOrCreateEntryByPath(fileName: string, path: Path): HostFileInformation {
-            const info = this.getEntryByPath(path) || this.createEntry(fileName, path);
-            return isString(info) ? undefined! : info; // TODO: GH#18217
+        public getOrCreateEntryByPath(fileName: string, path: Path, isAuxiliaryFile: boolean): HostFileInformation {
+            let info = this.getEntryByPath(path) || this.createEntry(fileName, path, isAuxiliaryFile);
+            if (isString(info)) return undefined!; // TODO: GH#18217
+            if (isAuxiliaryFile !== info.isAuxiliaryFile) {
+                info = this.createEntry(fileName, path, isAuxiliaryFile);

```

Comment:
I think I need to revisit this after some recent changes. Ignore this `isAuxiliaryFile` in HostCache for now.